UnicodeJsps/src/main/java/org/unicode/jsp/UnicodeJsp.java

```diff
@@ -208,28 +208,39 @@ public static void getDifferences(
         UnicodeUtilities.getDifferences(setA, setB, abbreviate, abResults, abSizes, abLinks);
     }
 
-    public static int parseCode(String text, String nextButton, String previousButton) {
+    public static int[] parseCode(String text, String nextButton, String previousButton) {
         // text = fromHTML.transliterate(text);
         String trimmed = text.trim();
-        if (trimmed.length() > 1) {
+        // Accept U+ notation and standalone hexadecimal digits, as well as a variety
+        // of hexadecimal character escape and numeric literal syntaxes.
+        Matcher matcher =
+                Pattern.compile(
+                                "(?:U\\+|\\\\[ux]\\{?|&#x|0x|16#|&H)?([0-9a-f'_]+)[\\};#]?",
```

Comment:
Ideally this should go into a separate tool over in unicodetools/src/java/org/unicode somewhere and have its own tests. could be future.